lang/c++/impl/json/JsonIO.hh

```diff
@@ -404,9 +404,25 @@ public:
 
     template<typename T>
     void encodeNumber(T t) {
```

Comment:
Instead of having a template function `encodeNumber` and its specializations, it'll be cleaner to have (a) a template function `numberToString` and its specialization to convert the given number into a string and then (b) have a single function `encodeNumber` that puts the string into `out_`.